src/garage/sampler/worker.py

```diff
@@ -0,0 +1,215 @@
+            agent(Policy or None): The worker's initial agent.
+            env(gym.Env or None): The worker's environment.
+        """
+
+    def update_agent(self, agent_update):
+        """Update the worker's agent, using agent_update."""
+
+    def update_env(self, env_update):
+        """Update the worker's env, using env_update."""
+
+    def rollout(self):
+        """Sample a single rollout of the agent in the environment.
+
+        Returns:
+            observations(np.array): Non-flattened array of observations.
+            actions(np.array): Non-flattened array of actions.
+            rewards(np.array): Array of rewards of shape (timesteps, 1).
+            agent_infos(Dict[str, np.array]): Dictionary of stacked,
+                non-flattened `agent_info`s.
+            env_infos(Dict[str, np.array]): Dictionary of stacked,
+                non-flattened `env_info`s.
+
+        """
+
+    def __getstate__(self):
+        """Refuse to be pickled."""
+        raise ValueError('Workers are not pickleable. '
+                         'Please pickle the WorkerFactory instead.')
+
+
+class DefaultWorker(Worker):
+    """Initialize a worker.
+
+    Args:
+        seed(int): The seed to use to intialize random number generators.
+        max_path_length(int): The maximum length paths which will be sampled.
+        worker_number(int): The number of the worker this update is
+            occurring in. This argument is used to set a different seed for
+            each worker.
+
+    Should create the following fields:
+        agent(Policy or None): The worker's initial agent.
+        env(gym.Env or None): The worker's environment.
+    """
+
+    def __init__(
+            self,
+            *,  # Require passing by keyword, since everything's an int.
+            seed,
+            max_path_length,
+            worker_number):
+        self.agent = None
+        self.env = None
+        self._seed = seed
+        self._max_path_length = max_path_length
+        self._worker_number = worker_number
+        self._observations = []
+        self._actions = []
+        self._rewards = []
+        self._agent_infos = defaultdict(list)
+        self._env_infos = defaultdict(list)
+        self._last_obs = None
+        self._path_length = 0
+        self.worker_init()
+
+    def worker_init(self):
+        """Initialize a worker.
+
+        Args:
+            worker_number(int): The number of the worker this update is
+                occurring in. This argument is used to set a different seed for
+                each worker.
+        """
+        deterministic.set_seed(self._seed + self._worker_number)
+
+    def update_agent(self, agent_update):
+        """Update an agent, assuming it implements garage.tf.policies.Policy.
+
+        Args:
+            agent_update(Dict[str, np.array]): Parameters to agent, which
+                should have been generated by calling
+                `policy.get_param_values`.
```

Comment:
i think Anson was planning on deleting `get_param_values` since `Parameterized` no longer exists. I guess this will be whatever ends up in the interface `garage.Policy`